libraries/chain/include/eosio/chain/log_catalog.hpp

```diff
@@ -59,13 +59,13 @@ struct log_catalog {
 
    uint32_t first_block_num() const {
       if (empty())
-         return 0;
+         return std::numeric_limits<block_num_t>::max();
       return collection.begin()->first;
    }
 
    uint32_t last_block_num() const {
       if (empty())
-         return 0;
+         return std::numeric_limits<block_num_t>::max();
```

Comment:
I think that one should be `min()` logically.

Reply:
And because the function returns `uint32_t`, it should be `std::numeric_limits<uint32_t>::min();`

Reply:
I am confused, I changed it to use max() based on your request so the line `if (collection.empty() || block_num < collection.begin()->first)` can become `if (block_num < collection.first_block_num())`. Do you want to revert that back by using `min()` instead?

Reply:
Well, I feel that the `last_block_num()` should return the smallest possible value if no blocks are present, and the `first_block_num()` should return the largest possible value if no blocks are present.